src/aes.c

```diff
@@ -365,48 +365,129 @@ int aes_ecb_encrypt (unsigned char *out, const unsigned char *in, aes_context_t
 }
 
 
-#define fix_xor(target, source) *(uint32_t*)&(target)[0] = *(uint32_t*)&(target)[0] ^ *(uint32_t*)&(source)[0]; *(uint32_t*)&(target)[4] = *(uint32_t*)&(target)[4] ^ *(uint32_t*)&(source)[4]; \
-                                *(uint32_t*)&(target)[8] = *(uint32_t*)&(target)[8] ^ *(uint32_t*)&(source)[8]; *(uint32_t*)&(target)[12] = *(uint32_t*)&(target)[12] ^ *(uint32_t*)&(source)[12];
-
-
 int aes_cbc_encrypt (unsigned char *out, const unsigned char *in, size_t in_len,
                      const unsigned char *iv, aes_context_t *ctx) {
 
-  uint8_t tmp[AES_BLOCK_SIZE];
-  size_t i;
-  size_t n;
+  size_t n = in_len / 16; // number of blocks
+  in_len &= 15;           // remaining bytes
+
+  __m128i ivec = _mm_loadu_si128((__m128i*)iv);
+
+  while(n) {
+    __m128i tmp = _mm_loadu_si128((__m128i*)in);
+    in += 16; n--;
+    tmp = _mm_xor_si128(tmp, ivec);
+
+    tmp = _mm_xor_si128       (tmp, ctx->rk_enc[ 0]);
```

Comment:
What do yoy think about using a loop?

Reply:
Haven't thought about those for a while!  – Will replace.